pkg/types/gcp/validation/machinepool.go

```diff
@@ -80,3 +84,45 @@ func ValidateDefaultDiskType(p *gcp.MachinePool, fldPath *field.Path) field.Erro
 
 	return allErrs
 }
+
+func ValidateKMSKey(platform *gcp.Platform, p *types.MachinePool, fldPath *field.Path) field.ErrorList {
```

Comment:
This validation should go in the validation run by pkg/asset. pkg/types should include minimal external dependencies to allow easy importing by other packages. The pkg/asset validation is called here: https://github.com/openshift/installer/blob/d2bb75079b1d73ed533c3b998550090893237bd5/pkg/asset/installconfig/gcp/validation.go#L52-L75